cli/tests/test_ci_workflow_efficiency.py

```
@@ -51,6 +52,26 @@ def test_ci_shards_python_once_and_does_not_repeat_unified_corpus() -> None:
     assert 'name = "pytest-xdist"' in lock
 
 
+def test_ci_preserves_make_test_context_without_repeating_test_work() -> None:
+    workflow_text = (ROOT / ".github/workflows/ci.yml").read_text(encoding="utf-8")
+    jobs = yaml.load(workflow_text, Loader=yaml.BaseLoader)["jobs"]
```

Ruff (0.16.1)

[error] Probable use of unsafe loader `BaseLoader` with `yaml.load`. Allows instantiation of arbitrary objects. Consider `yaml.safe_load`.

(S506)